File: football-ticket-ui/src/app/share/model/MatchDetailDTO.ts
export class MatchDetailDTO {
  constructor(
    public matchId: number = 0,
    public homeTeam: string = '',
    public awayTeam: string = '',
    public homeLogo: string = '',
    public awayLogo: string = '',
    public homeGoal: number = 0,
    public awayGoal: number = 0,
    public matchDate: string = '',
    public matchTime: string = '',
    public leagueName: string = '',
    public stadium: string = '',
    public stadiumImage: string = '',
    public matchStatus: string = '',
    public types: MatchType[] = [],
  ) {
  }
}


export class MatchType {
  constructor(
    public typeId: number = 0,
    public name: string = '',
    public description: string = '',
    public price: number = 0,
    public seats: MatchSeat[] = [],
  ) {
  }
}

export class MatchSeat {
  constructor(
    public seatId: number = 0,
    public seatNumber: string = '',
  ) {
  }
}

File: football-ticket-service/src/main/java/com/app/footballticketservice/dto/MatchDetailDTO.java
package com.app.footballticketservice.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Date;
import java.sql.Time;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class MatchDetailDTO {
    private Long matchId;
    private String homeTeam;
    private String awayTeam;
    private String homeLogo;
    private String awayLogo;
    private Integer homeGoal;
    private Integer awayGoal;
    private Date matchDate;
    private Time matchTime;
    private String leagueName;
    private String stadium;
    private String stadiumImage;
    private String matchStatus;
}
